### DISCUSSION TITLE
`piccolo user list` command?

### DISCUSSION THREAD

**Discussion by waldner** [piccolo-orm/piccolo_admin#360] [Ideas]
It would be useful to list existing users (for example, to avoid creating the admin user again if it exists; useful eg when running init containers in kbernetes).

**Comment by dantownsend** [comment]
It would be quite easy to add:

```python
from piccolo.apps.user.tables import BaseUser

async def list_users():
    usernames = await BaseUser.select(BaseUser.username).output(as_list=True)
    print('\n'.join(usernames))
```

If you wanted you could add it as a custom command to your own [app](https://piccolo-orm.readthedocs.io/en/latest/piccolo/projects_and_apps/piccolo_apps.html?highlight=command#commands).

It might be worth adding to Piccolo itself as a `piccolo user list` command.

An alternative is to modify the `piccolo user create` command, so it has an `if_not_exists` option.

**Reply by waldner** [reply]
What about https://github.com/piccolo-orm/piccolo/pull/921 ?

**Reply by waldner** [reply]
Well actually wait as I want to add more info to the output, while we're at it.

**Reply by waldner** [reply]
Ok, ready for review. Not sure how to deal with coverage, shall I add a test that creates a user and lists it?
